docs/user_guide/backend_configuration.rst

```diff
@@ -0,0 +1,304 @@
+
+.. code-block:: python
+
+    from uuid import uuid4
+    from datetime import datetime
+
+    from dateutil import tz
+    from neuroconv.tools.nwb_helpers import make_or_load_nwbfile, get_default_backend_configuration, configure_backend
+    from pynwb import TimeSeries
+
+    nwbfile_path = "./my_nwbfile.nwb"
+    backend="hdf5"
+
+    session_start_time = datetime(2020, 1, 1, 12, 30, 0, tzinfo=tz.gettz("US/Pacific"))
+    nwbfile = pynwb.NWBFile(
+        session_start_time=session_start_time,
+        session_description="My description...",
+        identifier=str(uuid4()),
+    )
+
+    # Add neurodata objects to the NWBFile, for example...
+    time_series = TimeSeries(
+        name="MyTimeSeries",
+        description="A time series from my experiment.",
+        unit="cm/s",
+        data=[1., 2., 3.],
+        timestamps=[0.0, 0.2, 0.4],
+    )
+    nwbfile.add_acquisition(time_series)
+
+    with make_or_load_nwbfile(
+        nwbfile_path=nwbfile_path,
+        nwbfile=nwbfile,
+        overwrite=True,
+        backend=backend,
+        verbose=True,
+    ):
+        backend_configuration = get_default_backend_configuration(
+            nwbfile=nwbfile, backend=backend
+        )
+
+        # Make any modifications to the configuration in this step, for example...
+        backend_configuration["acquisition/MyTimeSeries/data"].compression_options = dict(level=7)
+
+        configure_backend(
+            nwbfile=nwbfile, backend_configuration=backend_configuration
+        )
+
+
+
+FAQ
+---
+
+**How do I see what compression methods are available on my system?**
+
+You can see what compression methods are available on your installation by printing out the following variable:
+
+    .. code-block:: python
+
+      from neuroconv.tools.nwb_helpers import AVAILABLE_HDF5_COMPRESSION_METHODS
+
+      AVAILABLE_HDF5_COMPRESSION_METHODS
+
+    .. code-block:: bash
+
+      {'gzip': 'gzip',
+       ...
+       'Zstd': hdf5plugin._filters.Zstd}
+
+    And likewise for ``AVAILABLE_ZARR_COMPRESSION_METHODS``.
+
+**Can I modify the maximum shape or data type through the NeuroConv backend configuration?**
+
+Core fields such as the maximum shape and data type of the source data cannot be altered using the NeuroConv backend configuration.
+
+Instead, they would have to be changed at the level of the read operation; these are sometimes exposed to the initialization inputs or source data options.
+
+
+**Can I specify a buffer shape that incompletely spans the chunks?**
+
+The ``buffer_shape`` must be a multiple of the ``chunk_shape`` along each axis.
+
+This was found to give significant performance increases compared to previous data iterators that caused repeated I/O operations through partial chunk writes.
+
+**How do I disable chunking and compression completely?**
+
+To completely disable chunking (i.e., 'contiguous' layout), set both ``chunk_shape=None`` and ``compression_method=None``.
```

Comment:
```suggestion
To completely disable chunking for HDF5 backends (i.e., 'contiguous' layout), set both ``chunk_shape=None`` and ``compression_method=None``.
```

Reply:
Maybe not that important and this is what you meant but you can pass  `chunks=None` to zarr and get all the data in one file:

```
import zarr

# Create a Zarr group
zarr_folder = zarr.open_group('test_zarr.zarr', mode='w')

# Create an array and add it to the Zarr group
array_data = np.random.rand(20, 20, 20)
zarr_array = zarr_folder.create_dataset('array', data=array_data, chunks=None)
zarr_array.info
```

<table class="zarr-info"><tbody><tr><th style="text-align: left">Name</th><td style="text-align: left">/array</td></tr><tr><th style="text-align: left">Type</th><td style="text-align: left">zarr.core.Array</td></tr><tr><th style="text-align: left">Data type</th><td style="text-align: left">float64</td></tr><tr><th style="text-align: left">Shape</th><td style="text-align: left">(20, 20, 20)</td></tr><tr><th style="text-align: left">Chunk shape</th><td style="text-align: left">(20, 20, 20)</td></tr><tr><th style="text-align: left">Order</th><td style="text-align: left">C</td></tr><tr><th style="text-align: left">Read-only</th><td style="text-align: left">False</td></tr><tr><th style="text-align: left">Compressor</th><td style="text-align: left">Blosc(cname='lz4', clevel=5, shuffle=SHUFFLE, blocksize=0)</td></tr><tr><th style="text-align: left">Store type</th><td style="text-align: left">zarr.storage.DirectoryStore</td></tr><tr><th style="text-align: left">No. bytes</th><td style="text-align: left">64000 (62.5K)</td></tr><tr><th style="text-align: left">No. bytes stored</th><td style="text-align: left">56455 (55.1K)</td></tr><tr><th style="text-align: left">Storage ratio</th><td style="text-align: left">1.1</td></tr><tr><th style="text-align: left">Chunks initialized</th><td style="text-align: left">1/1</td></tr></tbody></table>

Reply:
I did not check myself (trusted @bendichter on that one)

Our config literally just passes the values on to downstream calls so I assume this would work here, in which case we can remove this line as a minor point (even though `chunks=None` is the option, and even though the array says 1/1 chunks, that's all 'equivalent' to contiguous layout in a sense)